tcms/testcases/tests/test_views.py

```diff
@@ -331,6 +331,107 @@ def test_edit_a_case(self):
+            default_tester=self.tester,
+            reviewer=self.tester,
+            case_status=self.case_status_proposed,
+            plan=[self.plan],
+        )
+        response = self.client.get(reverse("testcases-edit", args=[testcase.pk]))
+        self.assertContains(
+            response,
+            '<input type="text" id="id_default_tester" name="default_tester" '
+            f'value="{self.tester.email}" class="form-control">',
+        )
+
+
+class TestEditTestCaseViewPermission(PermissionsTestCase):
+    permission_label = "testcases.change_testcase"
+    http_method_names = ["get", "post"]
+    url = reverse(
+        "testcases-edit", args=[0]
+    )  # Workaround for passing check_mandatory_attributes
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.summary = "updated summary"
+        cls.post_data = {
+            "summary": cls.summary,
+        }
+        super().setUpTestData()
+        user_should_have_perm(cls.tester, "testcases.view_testcase")
+
+        case_status_confirmed = TestCaseStatus.objects.filter(is_confirmed=True).first()
+
+        cls.case = TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=case_status_confirmed,
+        )
+        cls.url = reverse("testcases-edit", args=[cls.case.pk])
+        # case.save()  # creates a new case entry in the database
+
+        cls.post_data.update(
+            {
+                "author": cls.tester.pk,
+                "default_tester": cls.tester.pk,
+                "priority": cls.case.priority.pk,
+                "product": cls.case.category.product.pk,
+                "category": cls.case.category.pk,
+                "case_status": case_status_confirmed.pk,
+                "text": "some text description",
+                "script": "some script",
+                "arguments": "args1, args2, args3",
+                "email_settings-0-auto_to_case_author": "on",
+                "email_settings-0-auto_to_run_manager": "on",
+                "email_settings-0-auto_to_execution_assignee": "on",
+                "email_settings-0-auto_to_case_tester": "on",
+                "email_settings-0-auto_to_run_tester": "on",
+                "email_settings-0-notify_on_case_update": "on",
+                "email_settings-0-notify_on_case_delete ": "on",
+                "email_settings-0-cc_list": "info@example.com",
+                "email_settings-0-case": "",
+                "email_settings-0-id": cls.case.emailing.pk,
+                "email_settings-TOTAL_FORMS": "1",
+                "email_settings-INITIAL_FORMS": "1",
+                "email_settings-MIN_NUM_FORMS": "0",
+                "email_settings-MAX_NUM_FORMS": "1",
+            }
+        )
+
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, _("Edit TestCase"))
```

Comment:
Should assert that the page we're seeing is editing exactly the same object we think it is.

Reply:
@atodorov this test class is to check permissions. Isn't making such assertions out of scope of this class?

Reply:
Is that already checked somewhere else?

Reply:
Wouldn't this test suffice it? 
https://github.com/kiwitcms/Kiwi/blob/5293fa3102336a82956e2fdc776e98ad5a5fb90a/tcms/testcases/tests/test_views.py#L321-L332

Reply:
No. That's a POST which redirects to the page showing the results. You are trying to open the Edit page and need to verify that it is indeed editing the object which you want (whose ID is in the URL).

Reply:
Added such tests here: https://github.com/kiwitcms/Kiwi/pull/2257/files#diff-6ae2ec47dc4f60776d7dd22884bad393555fdf14e8fb6f2e620838bb372170e3R319

Reply:
Nope you haven't. These are all different scenarios and they are asserting on different conditions.